src/pdf.rs

```diff
@@ -0,0 +1,673 @@
+use crate::database::set_up_database;
+use crate::models::Worktime;
+use crate::service::worktime;
+use chrono::{DateTime, Datelike, Local, NaiveDate};
+use printpdf::*;
+use sqlx::postgres::types::PgInterval;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::BufWriter;
+
+fn add_month(given_month: &str) -> String {
+    let year: i32 = given_month[0..4].parse().unwrap();
+    let month: u32 = given_month[5..7].parse().unwrap();
+
+    let new_month = if month == 12 { 1 } else { month + 1 };
+
+    let new_year = if month == 12 { year + 1 } else { year };
+    format!("{:04}-{:02}", new_year, new_month)
+}
+
+fn get_weekday_abbreviation(date: chrono::NaiveDate) -> String {
+    match date.weekday() {
+        chrono::Weekday::Mon => "Mo".to_string(),
+        chrono::Weekday::Tue => "Di".to_string(),
+        chrono::Weekday::Wed => "Mi".to_string(),
+        chrono::Weekday::Thu => "Do".to_string(),
+        chrono::Weekday::Fri => "Fr".to_string(),
+        chrono::Weekday::Sat => "Sa".to_string(),
+        chrono::Weekday::Sun => "So".to_string(),
+    }
+}
+
+fn add_durations(d1: &PgInterval, d2: &PgInterval) -> PgInterval {
+    PgInterval {
+        months: d1.months + d2.months,
+        days: d1.days + d2.days,
+        microseconds: d1.microseconds + d2.microseconds,
+    }
+}
+
+fn generate_schedule(worktimes: Vec<Worktime>, year: i32, month: u32) -> Vec<Vec<String>> {
+    let mut schedule: Vec<Vec<String>> = Vec::new();
+
+    // Determine the number of days in the month
+    let num_days = NaiveDate::from_ymd_opt(year, month, 1)
+        .and_then(|date| date.with_month(month + 1))
+        .map(|next_month| (next_month - chrono::Duration::days(1)).day())
+        .unwrap_or(31);
+
+    for day in 1..=num_days {
+        if let Some(date) = NaiveDate::from_ymd_opt(year, month, day) {
+            let mut day_entry = vec![get_weekday_abbreviation(date)];
+            let mut total_duration = PgInterval {
+                months: 0,
+                days: 0,
+                microseconds: 0,
+            };
+
+            // Filter worktimes for the current day
+            for worktime in worktimes
+                .iter()
+                .filter(|w| w.start_time.date_naive() == date)
+            {
+                let task_description = worktime.task_id;
```

Comment:
query the task description with the task_id
comment with TODO if its not implemented yet